### PR TITLE
Update to unified automation sdk v1.7.1.

### PR DESCRIPTION
**Pull request by bveldhoen**
Breaking change for pyUAF clients:
- Removal of (in ua sdk) deprecated constructor for PkiCertificate,
replaced by two other constructors.

### CODE REVIEW

File: src/UafMacros.cmake [hunk 1]
```diff
@@ -215,6 +214,26 @@ MACRO(handleUnifiedAutomationSdk)
         message(WARNING "The demo server (UaServerCPP) cannot be found: ${DEMOSERVER_COMMAND} does not exist! This demo server is only needed to run the client unit tests, so you can safely ignore this warning if you do not intend to run unit tests.\n")
     endif (EXISTS ${DEMOSERVER_COMMAND})
 
+    set(UASDK_LIBRARIES_DIRS
+        "${UASDK}/lib"
+        "${UASDK}/lib64")
+    set(UASDK_INCLUDE_DIRS
+        SYSTEM "${UASDK}/include"
```
Comment (bveldhoen) [inline src/UafMacros.cmake]: I could change this (and below) into "${UASDK_INCLUDE_DIR}".

File: src/UafMacros.cmake [hunk 2]
```diff
@@ -183,11 +182,11 @@ MACRO(handleUnifiedAutomationSdk)
     endif (EXISTS "${UASDK_DIR}/src")
     
     # figure out if the SDK version is at least 1.4 by checking if include/uabase/uafile.h exists
-    if (EXISTS "${UASDK_INCLUDE_DIR}/uabase/uafile.h")
+    if (EXISTS "${UASDK_INCLUDE_DIR}/uabasecpp/uafile.h")
```
Comment (bveldhoen) [inline src/UafMacros.cmake]: Some ua sdk folders were renamed with a suffix of "cpp".

File: src/UafMacros.cmake [hunk 3]
```diff
@@ -215,6 +214,26 @@ MACRO(handleUnifiedAutomationSdk)
         message(WARNING "The demo server (UaServerCPP) cannot be found: ${DEMOSERVER_COMMAND} does not exist! This demo server is only needed to run the client unit tests, so you can safely ignore this warning if you do not intend to run unit tests.\n")
     endif (EXISTS ${DEMOSERVER_COMMAND})
 
+    set(UASDK_LIBRARIES_DIRS
+        "${UASDK}/lib"
```
Comment (bveldhoen) [inline src/UafMacros.cmake]: uastack was built to lib, the rest to lib64.

File: src/UafMacros.cmake [hunk 4]
```diff
@@ -215,6 +214,26 @@ MACRO(handleUnifiedAutomationSdk)
...
+    set(UASDK_INCLUDE_DIRS
+        SYSTEM "${UASDK}/include"
+        SYSTEM "${UASDK}/include/uabasecpp"
+        SYSTEM "${UASDK}/include/uaclientcpp"
+        SYSTEM "${UASDK}/include/uamodels"
+        SYSTEM "${UASDK}/include/uapkicpp"
+        SYSTEM "${UASDK}/include/uaservercpp"
+        SYSTEM "${UASDK}/include/uastack"
+        SYSTEM "${UASDK}/include/xmlparsercpp")
+    set(UASDK_LIBRARIES
```
Comment (bveldhoen) [inline src/UafMacros.cmake]: All libraries are now available through 1 cmake variable (UASDK_LIBRARIES).

File: src/UafMacros.cmake [hunk 5]
```diff
@@ -215,6 +214,26 @@ MACRO(handleUnifiedAutomationSdk)
...
+        debug uabasecppd optimized uabasecpp
+        debug uaclientcppd optimized uaclientcpp
+        debug uamodelsd optimized uamodels
+        debug uapkicppd optimized uapkicpp
+        debug uastackd optimized uastack
+        debug xmlparsercppd optimized xmlparsercpp)
```
Comment (bveldhoen) [inline src/UafMacros.cmake]: Note: uaservercpp is not included in UASDK_LIBRARIES, as this isn't used by pyUAF.

File: src/uaf/util/genericstructurevalue.cpp [hunk 6]
```diff
@@ -23,485 +23,450 @@
...
-		const StructureDefinition& structureDefinition)
-	{
-		UaByteString uaData;
-		UaStructureDefinition uaDef;
-		UaAbstractGenericValue::Encoding uaEnc = static_cast<UaAbstractGenericValue::Encoding>(encoding);
-		data.toSdk(uaData);
-		structureDefinition.toSdk(uaDef);
-		return SdkStatus( uaGenericStructureValue_.setGenericValue(uaData, uaEnc, uaDef) );
-	}
+using namespace uaf;
```
Comment (bveldhoen) [inline src/uaf/util/genericstructurevalue.cpp]: The formatting of this file was changed; I'll try and undo the formatting-only changes.

File: src/uaf/util/pkicertificate.cpp [hunk 7]
```diff
@@ -38,16 +38,34 @@ namespace uaf
     // Constructor
     // =============================================================================================
     PkiCertificate::PkiCertificate(
```
Comment (bveldhoen) [inline src/uaf/util/pkicertificate.cpp]: This constructor was already deprecated in ua sdk v1.5.5. It was removed in at least v1.7.1 (and perhaps already in an earlier version).

File: src/uaf/util/pkicertificate.cpp [hunk 8]
```diff
@@ -38,16 +38,34 @@ namespace uaf
...
-            const PkiPrivateKey&        issuerPrivateKey)
+        const PkiCertificateInfo& info,
+        const PkiIdentity& subject,
+        const PkiRsaKeyPair& subjectKeyPair,
+        bool bCACert)
+    : uaCert_(info.toSdk(),
+              subject.toSdk(),
+              subjectKeyPair.uaPair_,
+              bCACert,
+              UaPkiCertificate::SignatureAlgorithm_Sha256)
```
Comment (bveldhoen) [inline src/uaf/util/pkicertificate.cpp]: For the moment, this supports only SignatureAlgorithm_Sha256. To support the other SignatureAlgorithms, probably a new (enum) type in the uaf namespace needs to be created.

File: src/uaf/util/structurefielddatatypes.cpp [hunk 9]
```diff
@@ -22,45 +22,51 @@
...
-        }
-
-
-        // UAF to SDK type
-        // =========================================================================================
-        UaStructureFieldDataType fromUafToSdk(StructureFieldDataType type)
-        {
-            return static_cast<UaStructureFieldDataType>(type);
-        }
+namespace structurefielddatatypes
```
Comment (bveldhoen) [inline src/uaf/util/structurefielddatatypes.cpp]: Formatting off again.

**Comment by WimPessemier** [comment]
Hi @bveldhoen, thanks for the efforts! 
I'm happy to merge (and give you more access rights if you want) but could you only update the changelog and the version.txt please?

**Comment by bveldhoen** [comment]
> Hi @bveldhoen, thanks for the efforts!
> I'm happy to merge (and give you more access rights if you want) but could you only update the changelog and the version.txt please?

Yes on all of the above (pls)!

**Comment by bveldhoen** [comment]
Forced updated the commit with the proposed changes, please have a look?

**Pull request merged**